scipy/integrate/_ivp/lsoda.py

```diff
@@ -168,10 +168,12 @@ def _dense_output_impl(self):
         iwork = self._lsoda_solver._integrator.iwork
         rwork = self._lsoda_solver._integrator.rwork
 
-        order = iwork[14]
+        order = iwork[13]
         h = rwork[11]
         yh = np.reshape(rwork[20:20 + (order + 1) * self.n],
                         (self.n, order + 1), order='F').copy()
+        if iwork[14] < order:
```

Comment:
Looks like it can be achieved simpler by assigning `h = rwork[10]` in this case. Am I right?

Reply:
Without having a detailed look, I do not think so. What is happening here is that only the last term needs rescaling, so changing `h` would not properly account for that. This last-term problem is likely a consequence of a mistake somewhere in the associated Fortran code.